src/components/BottomBanner.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useRef } from "react";
+import styles from "./BottomBanner.module.scss";
+import { CustomButton } from "./CustomButton";
+
+interface BottomBannerProps {
+  text: string;
+  buttonText: string;
+  onButtonClick: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+export const BottomBanner = ({
+  text,
+  buttonText,
+  onButtonClick,
+}: BottomBannerProps) => {
+  const bannerRef = useRef<HTMLDivElement | null>(null);
+
+  const focusHandler = (event: KeyboardEvent) => {    
```

Comment:
im not quire sure why you would need this?

Reply:
It's to to trap focus. A better solution would be to just use a `<dialog>`, but that's not supported everywhere yet. Or you could use a [library](https://github.com/focus-trap/focus-trap-react).

Reply:
I implemented it as @LeoTuet begged me to do so and forgot dialog exists nowadays. Also don't think the browser support is that bad

Reply:
@94726 I would say browser support is still poor. It is supported in Firefox 98 and in Safari for macOS/iOS 15.4, but they're so new that almost nobody uses them yet. According to [caniuse](https://caniuse.com/dialog), only 62% of German users use a browser with `<dialog>` support.

Reply:
Oh you're totally, didn't know it's possible to look at the usage at caniuse